logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/GetNotificationsUseCase.kt

```diff
@@ -117,24 +126,28 @@ class GetNotificationsUseCaseImpl(
         selfUser: SelfUser,
         conversationMutedStatus: MutedConversationStatus
     ): Boolean =
-        when(conversationMutedStatus) {
-            MutedConversationStatus.AllAllowed -> true
-            MutedConversationStatus.OnlyMentionsAllowed -> {
-                when(val content = message.content) {
-                    is MessageContent.Text ->  {
-                        val containsSelfUserName = selfUser.name?.let { selfUsername ->
-                            content.value.contains("@$selfUsername")
-                        } ?: false
-                        val containsSelfHandle = selfUser.handle?.let { selfHandle ->
-                            content.value.contains("@$selfHandle")
-                        } ?: false
-
-                        containsSelfUserName or containsSelfHandle
+        when {
+            message is Message.Server -> false
+            message.visibility != Message.Visibility.VISIBLE -> false
```

Comment:
Maybe extract to `shouldShowMessage(message)` function?

Reply:
Sure, done 👍